kong/kong.lua

```diff
@@ -252,6 +252,18 @@ function Kong.balancer()
   end
 end
 
+function Kong.rewrite()
+  core.rewrite.before()
+
+  for _, plugin in ipairs(singletons.loaded_plugins) do
+    if plugin.schema.no_consumer and plugin.schema.no_api then
+      plugin.handler:rewrite()
+    end
+  end
+
+  core.rewrite.after()
+end
+
```

Comment:
hence if you run only the plugins that have been globally added by the user, neither attached to a consumer nor an api, it would work. And it would be up to the user to whether to add this plugin or not.